src/pages/Admin/Venue/Details/VenueDetails.tsx

```diff
@@ -0,0 +1,254 @@
+      if (owners && owners.length > 0) {
+        for (const owner of owners) {
+          const user = await firebase
+            .functions()
+            .httpsCallable("venue-getOwnerData")({ userId: owner });
+
+          const userData = user.data;
+
+          if (ownersRef.current.filter((i: Owner) => i.id !== owner)) {
+            newOwners.push({
+              id: owner,
+              ...userData,
+            });
+          }
+        }
+
+        if (!isEqual(ownersData, newOwners)) {
+          setOwnersData(newOwners);
+        }
+      }
+    }
+    getOwnersData();
+  }, [owners, ownersData]);
+
+  if (!user) return null;
+
+  const toggleRoomModal = () => setModalOpen(!modalOpen);
+
+  const handleEditRoom = (room: RoomData_v2, index: number) => {
+    setEditingRoom({
+      ...room,
+      roomIndex: index,
+    });
+  };
+
+  const handleRoomEvent = (roomName: string) => {
+    setEventRoom(roomName);
+    setShowEventModal(true);
+  };
+
+  const handleEditRoomSave = (values: RoomData_v2, index: number) => {
+    const newData = {
+      ...values,
+      x_percent: editingRoom?.x_percent,
+      y_percent: editingRoom?.y_percent,
+      width_percent: editingRoom?.width_percent,
+      height_percent: editingRoom?.height_percent,
+    };
+
+    updateRoom(newData, venueId!, user, index);
+  };
+
+  return (
+    <S.Container>
+      <S.Header>
+        <VenueHero
+          bannerImageUrl={coverImageUrl}
+          logoImageUrl={icon}
+          name={name}
+          subtitle={subtitle}
+          description={description}
+          large
+          showEdit
+          venueId={venue.id!}
+        />
+
+        <S.HeaderActions>
+          <Link
+            to={`/in/${venue.id}`}
+            className="btn btn-primary"
+            style={{ marginBottom: "0.5em" }}
+            target="_blank"
+          >
+            Go to your space
+          </Link>
+          <Button>Preview landing page</Button>
+
+          <S.AdminList>
+            <S.AdminListTitle>Party admins</S.AdminListTitle>
+
+            {ownersData.length > 0 &&
+              ownersData.map((owner: Owner) => (
+                <S.AdminItem key={owner.id}>
+                  <S.AdminPicture backgroundImage={owner.pictureUrl} />
+                  <S.AdminItemTitle>{owner.partyName}</S.AdminItemTitle>
+                </S.AdminItem>
+              ))}
+
+            <Button onClick={() => setShowOwnersModal(true)}>
+              Invite an admin
+            </Button>
+          </S.AdminList>
+        </S.HeaderActions>
+      </S.Header>
+
+      <S.Main>
+        <MapPreview
+          venueId={venueId!}
+          venueName={name}
+          mapBackground={mapBackgroundImageUrl}
+          rooms={rooms}
+        />
+
+        {!!mapBackgroundImageUrl && (
+          <>
+            <S.RoomActions>
+              <S.RoomCounter>{rooms ? rooms.length : "0"} Rooms</S.RoomCounter>
+              <Button onClick={() => toggleRoomModal()} gradient>
```

Comment:
Function + useCallback.

Reply:
In this instance, since there are no parameters that need to be passed to `toggleRoomModal`, you can actually just do:

```suggestion
              <Button onClick={toggleRoomModal} gradient>
```